### DISCUSSION TITLE
Many clients on my local network ?

### DISCUSSION THREAD

**Discussion by ins0u** [AdguardTeam/AdGuardHome#5622] [General]
Hi,

I think I have a problem.

When I am on the home page, in "Best customers", I notice that I have a lot of customers connected.

Many of them are outside my local network (???)

![client](https://user-images.githubusercontent.com/12513250/226980018-085d8307-73ca-4bfe-a9b0-ae2f489eb9eb.png)

When I update the logs, I always have a lot of requests that go to sites that I don't know, by ips (clients) that I don't know either.

![client2](https://user-images.githubusercontent.com/12513250/226980027-2147bca3-5e65-4d73-8f84-0428932bc68d.png)

I feel like I have a device on my local network that is infected and sending requests everywhere.

Do you agree with this analysis ?
If so, how can I stop all these requests ?

Thx for your helps :)

**Comment by ppfeufer** [comment]
That means that your AGH instance can be reached from the internet and some are using it.

You should restrict access to it only for your home network.

**Comment by ins0u** [comment]
`You should restrict access to it only for your home network.`

How to do that?
Sorry it's new for me 😅

But normally, there is a login/password to prevent that, right?  Does that also mean it was found?

Thx for your help :)

**Reply by ppfeufer** [reply]
The Login/Password is only for the web UI.
If your AGH instance is publically accessible, everyone can use it as their DNS resolver.

You can limit this in AGH's config file `/opt/AdGuardHome/AdGuardHome.yaml`.
Here you'll find a setting called `bind_host` which is the network interface AGH is listening to. Set this to the internal IP address of the device your AGH instance is running on.

Example:
```yaml
dns:
  bind_hosts:
    - 10.10.10.2
    - fd00::2
```

As you can see, here are 2 IP addresses listed. Both are the internal IPv4 and IPv6 addresses of AGH's device.

Important:
Before you change any setting in this file stop AGH, otherwise, your changes might be overridden when AGH starts again. After you changed the setting, start AGH again.

**Reply by ins0u** [reply]
OK, I understand a little better now.

However, I can't find the configuration file:
`/opt/AdGuardHome/AdGuardHome.yaml`

![Screenshot_20230323_133908_RaspController](https://user-images.githubusercontent.com/12513250/227206741-bf9c7514-9c65-4b8c-b8ea-1ebb69ddc4d0.jpg)

Maybe, my configuration file is an another place ?

**Reply by ppfeufer** [reply]
Hmm, that's odd.
![image](https://user-images.githubusercontent.com/2989985/227208595-b25413d6-dced-4831-9570-fc3a28fa80ec.png)

How did you install AGH?
Maybe depending on the installation method the file might be in a different location. I suggest searching for it.

**Reply by ins0u** [reply]
OK, I found it :
`/home/pi/AdguardHome/AdGuardHome.yalm`

In my file :

```
dns:
  bind_hosts:
    - 0.0.0.0
```

So if I understand correctly, I have to replace the 0.0.0.0 with the ip of the machine where AdguardHome is installed ?

It is installed on a raspberry whose ip is 192.168.0.20

So :

- STOP AGH (cmd line ?) 
- Modify config file

```
dns:
  bind_hosts:
    - 192.168.0.20
```

- Restart AGH

Thx for your help 😁

**Reply by ppfeufer** [reply]
Exactly. `0.0.0.0` means it's listening to all IPs, including the public IPv6, which explains how others can use your AGH instance from the outside.
You might want to add the device's internal IPv6 there as well if there is one. They usually start with `fd00:`

Make sure that the exact same IPs are used in the DNS settings of your devices / DHCP server / Router.

**Reply by ins0u** [reply]
Ok, i stopped AGH with : 
`sudo ./AdGuardHome -s stop`

Replace the config file with the new value

Re-start AGH : 
`sudo ./AdGuardHome -s start` and `sudo ./AdGuardHome -s reload` and reboot my raspberry but there are always requests coming from outside my network :(

I don't know why..

Have you an another idea ? ^^

**Reply by ppfeufer** [reply]
In your AGH web UI, click on "Setup Guide" and see on what IPs it's listening. Should only be the internal IPs and the domain name if one is set up. If there is a domain name set up, make sure it is not reachable from the outside.

**Reply by ins0u** [reply]
There is only the ip of the rapsberry (192.168.0.20)

**Reply by ppfeufer** [reply]
No domain name?

Then it should only be listening to requests from your own network.

**Reply by ins0u** [reply]
Nop, no domaine name..

Which domain names should there be ? :/

**Reply by ins0u** [reply]
I found an alternative solution.

In "DNS settings", "Access settings", "Authorized clients", I added all the ips of the machines in my local network.

Now in my logs, I have only the machines of my network.

But I have a question: does it block all the ips that are external ? or does it pass anyway but without going through the dns ?

The goal is to have ONLY the machines of my network that use internet.
